src/embedded.test.js

```diff
@@ -0,0 +1,608 @@
+const mockClientId = 'abcdef0123456789abcdef0123456789';
+
+describe('HelloSign', () => {
+
+  afterEach(() => {
+    // Clear DOM.
+    document.getElementsByTagName('html')[0].innerHTML = '';
+  });
+
+  test('is defined', () => {
+    expect(HelloSign).toBeDefined();
+  });
+
+  test('instantiates without errors', () => {
+    expect(() => new HelloSign()).not.toThrow();
+  });
+
+  test('throws when instantiated without an object', () => {
+    expect(() => new HelloSign(false)).toThrow(/must be an object/);
+  });
+
+  describe('accessors', () => {
+
+    describe('element()', () => {
+      test('returns the base element', () => {
+        const client = new HelloSign({ clientId: mockClientId });
+
+        client.open(mockSignURL);
+
+        expect(client.element).toBeInstanceOf(HTMLElement);
+
+        client.close();
+      });
+    });
+
+    describe('isOpen()', () => {
+      test('returns the open state', () => {
+        const client = new HelloSign({ clientId: mockClientId });
+
+        expect(client.isOpen).toEqual(false);
+
+        client.open(mockSignURL);
+
+        expect(client.isOpen).toEqual(true);
+
+        client.close();
+
+        expect(client.isOpen).toEqual(false);
+      });
+    });
+  });
+
+  describe('methods', () => {
+
+    describe('open()', () => {
+
+      test('closes the old window if embedded is already open', (done) => {
+        const client = new HelloSign({
+          clientId: mockClientId,
+        });
+
+        const fn = jest.fn(() => {
+          expect(fn).toBeCalledTimes(1);
+
+          client.close();
+
+          done();
+        });
+
+        client.once(HelloSign.events.CLOSE, fn);
+
+        client.open(mockSignURL);
+        client.open(mockSignURL);
+      });
+
+      test('appends markup into document body if no container is specified', () => {
+        const client = new HelloSign();
+        const expected = expect.stringContaining(settings.classNames.BASE);
+
+        client.open(mockSignURL, {
+          clientId: mockClientId,
+        });
+
+        expect(document.body.children.length).toBeGreaterThan(0);
+        expect(document.body.lastChild.className).toEqual(expected);
+
+        client.close();
+      });
+
+      test('appends markup into container if one is specified', () => {
+        const client = new HelloSign();
+        const container = document.createElement('div');
+        const expected = expect.stringContaining(settings.classNames.BASE);
+
+        client.open(mockSignURL, {
+          clientId: mockClientId,
+          container,
+        });
+
+        expect(container.children.length).toBeGreaterThan(0);
+        expect(container.firstChild.className).toEqual(expected);
+
+        client.close();
+      });
+
+      test('emits the "open" event', (done) => {
+        const client = new HelloSign();
+
+        const fn = jest.fn(() => {
+          expect(fn).toBeCalledTimes(1);
+
+          client.close();
+
+          done();
```

Comment:
Basically same issue with this assertion. It won't enforce anything because it can only be called once, and will not throw an error if it isn't called. I think in this case `done()` handles everything you need, and line 117 can be removed completely.

Reply:
Fixed